java.exercise/src/test/Sakila_Project.txt

```diff
@@ -0,0 +1,226 @@
+-- Create a new branch called f-sql
+-- Create a file called Sakila_Project.sql
+
+-- Sakila Database
+
+-- Write a SQL statement for each of the following. 
+
+-- 1. Get 10 cities in descending alphabetical order.
+
+select * 
+from sakila.city 
+order by city desc 
+limit 10;
+
+-- 2. Get all films with "airplane" in the title.
+
+select * 
+from sakila.film
+where title like '%airplane%';
+
+-- 3. Get the highest payment amount.
+
+select max(amount) 
+from sakila.payment;
+
+-- 4. Get the number of records in the customer table for store id #1.
+
+select count(*) 
+from sakila.customer 
+where store_id = 1;
+
+-- 5. Get all payment records for customer with email address "NANCY.THOMAS@sakilacustomer.org"
+
+select * 
+from sakila.payment p
+join sakila.customer c
+using ( customer_id )
+where c.email = 'NANCY.THOMAS@sakilacustomer.org';
+
+-- 6. Use a View to get the film info for actor Bob Fawcett.
+
+-- EITHER:
+select * 
+from sakila.film_list 
+where actors 
+like '%BOB FAWCETT%';
+-- returns a table
+
+-- OR:
+select film_info 
+from sakila.actor_info 
+where first_name = 'bob' 
+and last_name = 'fawcett';
+-- returns a string
+
+-- 7. Use a Stored Procedure to get the 4 inventory ids for the film "Alien Center" at Store #2. 
+
+set @var = 0;
+call sakila.film_in_stock(
+	(
+		select film_id 
+		from sakila.film 
+		where title = 'Alien Center'
+    ),
+	 2, 
+	 @var
+ );
+
+-- 8. Insert a new store. Ensure that you use TRANSACTION. (This one is possible but it's tough! Pay attention to constraints and the order that you are inserting data.) 
+
+start transaction;
+
+set @address = (
+	select address_id 
+	from sakila.address
+	where address_id not in (
+		select address_id 
+		from sakila.store
+	) 
+    limit 1
+);
+
+insert into sakila.staff(
+	first_name,	
+    last_name,
+	address_id, 
+    email,
+	store_id,
+	username, 
+    password
+	) values (
+    'trevor',
+    'kleinstuber',
+	@address, 
+    'tklein@email.site',
+	(
+		select store_id 
+		from sakila.store 
+		limit 1
+    ),
+	'tklein', 
+    'very secure password'
+	);
+    
+set @staff = LAST_INSERT_ID();
+
+insert into sakila.store(
+	manager_staff_id,
+    address_id
+    ) values (
+    @staff, 
+    @address
+    );
+
+set @mystore = LAST_INSERT_ID();
+
+update sakila.staff
+set store_id = @mystore
+where staff_id = @staff;
+
+-- 9. Update the timestamp to the current date/time for the new store you entered in the previous question. 
+
+update sakila.store
+set last_update = now()
+where store_id = @mystore;
```

Comment:
How do you know what @mystore will be?
Generalize this query so that it could be run at any time. Currently it required to be run after #8